src/Integration/MefServices/ActiveSolutionBoundTracker.cs

```diff
@@ -71,12 +75,43 @@ public ActiveSolutionBoundTracker(IHost host, IActiveSolutionTracker activeSolut
             this.ProjectKey = this.solutionBindingInformationProvider.GetProjectKey();
         }
 
+
         private void OnActiveSolutionChanged(object sender, EventArgs e)
         {
+            UpdateConnection();
+
             this.RaiseAnalyzersChangedIfBindingChanged();
             this.errorListInfoBarController.Refresh();
         }
 
+        private void UpdateConnection()
+        {
+            ISonarQubeService sqService = this.extensionHost.SonarQubeService;
+            if (sqService.IsConnected)
+            {
+                sqService.Disconnect();
+
+                if (this.extensionHost.ActiveSection?.DisconnectCommand.CanExecute(null) == true)
+                {
+                    this.extensionHost.ActiveSection?.DisconnectCommand.Execute(null);
+                }
+            }
+
+            bool isSolutionCurrentlyBound = this.solutionBindingInformationProvider.IsSolutionBound();
+
+            if (isSolutionCurrentlyBound)
+            {
+                var solutionBinding = this.extensionHost.GetService<ISolutionBindingSerializer>();
+                solutionBinding.AssertLocalServiceIsNotNull();
+                BoundSonarQubeProject boundProject = solutionBinding.ReadSolutionBinding();
+                var connectionInformation = boundProject.CreateConnectionInformation();
```

Comment:
There might be a neater way to get the connection information, but this works for now.